In [1]:
import pandas as pd

In [7]:
may = pd.read_pickle('../data/may.pkl')
may.head()

pubdatetime   latitude  longitude        sumdid  chargelevel  \
0 2019-05-01 00:01:41.247  36.136822 -86.799877  PoweredLIRL1         93.0   
1 2019-05-01 00:01:41.247  36.191252 -86.772945  PoweredXWRWC         35.0   
2 2019-05-01 00:01:41.247  36.144752 -86.806293  PoweredMEJEH         90.0   
3 2019-05-01 00:01:41.247  36.162056 -86.774688  Powered1A7TC         88.0   
4 2019-05-01 00:01:41.247  36.150973 -86.783109  Powered2TYEF         98.0   

   companyname  
0            0  
1            0  
2            0  
3            0  
4            0

In [8]:
may.tail()

pubdatetime  latitude  longitude  \
20292498 2019-05-31 23:59:42   36.1639   -86.7823   
20292499 2019-05-31 23:59:42   36.1606   -86.7768   
20292500 2019-05-31 23:59:42   36.1606   -86.7769   
20292501 2019-05-31 23:59:42   36.1624   -86.7745   
20292502 2019-05-31 23:59:42   36.1519   -86.7985   

                                               sumdid  chargelevel  \
20292498  Poweredbb60ef04-c719-597f-805f-20978964fd8c         49.0   
20292499  Powered9f5743ed-7f81-539b-9e37-6503d54445d3         68.0   
20292500  Powered5744d1db-3bbe-55e5-9a23-a93828ba54fc         53.0   
20292501  Poweredad1d6d3b-162a-5bec-bffd-cac4ac8dee1c         42.0   
20292502  Powered8505008e-97c9-5003-8975-2ea6b2634fa0         76.0   

          companyname  
20292498            5  
20292499            5  
20292500            5  
20292501            5  
20292502            5

In [4]:
may.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 20283582 entries, 0 to 20292502
Data columns (total 6 columns):
 #   Column       Dtype         
---  ------       -----         
 0   pubdatetime  datetime64[ns]
 1   latitude     float64       
 2   longitude    float64       
 3   sumdid       object        
 4   chargelevel  float64       
 5   companyname  int64         
dtypes: datetime64[ns](1), float64(3), int64(1), object(1)
memory usage: 1.1+ GB


#note - the entries and the index numbers are different because we removed bicycles from our dataset.  difference of 8920 entries, there's 8921 bicycles in the csv.

#i wonder how many individual scooters are in this dataset

In [5]:
may['sumdid'].value_counts() 

PoweredIE7H5RIN474PE    8921
PoweredUK65LOU3K5N7G    8748
PoweredIOHG2G3XKATUL    8576
PoweredJCKMQCAL2SSNJ    8521
Powered4NYK5PWADVIAU    8439
                        ... 
PoweredYEBSU               1
Powered523737              1
Powered3361332             1
Powered8019965             1
Powered1PKP7               1
Name: sumdid, Length: 8303, dtype: int64

#there's 8303 scooters across 20,283,582 rows of data
that's almost 2443 entries per scooter ... which i can see is not true.
max is 8921, min is 1

i wonder if there's a difference across companies as to how many times each scooter is entered
there's 7 companies - is SHOULD be something like 2,897,654 entries for each business if the distribution is equal

In [9]:
may['companyname'].value_counts()

3    7111655
0    4334006
1    3112581
5    2984630
4    1301001
2     915314
6     524395
Name: companyname, dtype: int64

the data is NOT equal
company 3 has a lot more info than anyone else
companies 1 and 5 are right about where i predicted
companies 0 and 4 are a little above and below
companies 2 and 6 are way underreported compared to the other 5.

In [10]:
may.loc[may['sumdid'] == 'PoweredLIRL1']
#company0 = may.loc[may['sumdid'] == 'PoweredLIRL1']
#note: Jeremiah's query to look at how often a scooter is pinged.  this one is every 5 mins

pubdatetime   latitude  longitude        sumdid  \
0        2019-05-01 00:01:41.247  36.136822 -86.799877  PoweredLIRL1   
2694     2019-05-01 00:06:41.537  36.136822 -86.799877  PoweredLIRL1   
26176    2019-05-01 01:01:43.917  36.136751 -86.799921  PoweredLIRL1   
28812    2019-05-01 01:06:44.237  36.136751 -86.799921  PoweredLIRL1   
30726    2019-05-01 01:11:44.167  36.136751 -86.799921  PoweredLIRL1   
...                          ...        ...        ...           ...   
20209588 2019-05-31 22:00:15.960  36.191117 -86.781360  PoweredLIRL1   
20213207 2019-05-31 22:05:16.080  36.191117 -86.781360  PoweredLIRL1   
20216631 2019-05-31 22:10:16.483  36.191096 -86.781369  PoweredLIRL1   
20220463 2019-05-31 22:15:16.653  36.191096 -86.781369  PoweredLIRL1   
20224154 2019-05-31 22:20:16.737  36.191122 -86.781343  PoweredLIRL1   

          chargelevel  companyname  
0                93.0            0  
2694             93.0            0  
26176            92.0            0  
28812            92.0            0  
30726            92.0            0  
...               ...          ...  
20209588         53.0            0  
20213207         53.0            0  
20216631         53.0            0  
20220463         53.0            0  
20224154         53.0            0  

[2859 rows x 6 columns]

so one of the scooters that had only 1 entry ... 

In [11]:
may.loc[may['sumdid']== 'Powered1PKP7']

pubdatetime   latitude  longitude        sumdid  \
16270541 2019-05-27 21:46:35.653  36.176051  -86.76609  Powered1PKP7   

          chargelevel  companyname  
16270541         82.0            0

same company ... interesting.

let me try and find out how many scooters each of these companies have.  
I know there's 8303 scooters and 7 companies.

In [ ]:
may['sumdid'].value_counts() 